test/integration/resourcemanager/seccompprofile/seccompprofile_suite_test.go

```diff
@@ -0,0 +1,124 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seccompprofile_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gardener/gardener/pkg/operation/botanist/component/resourcemanager"
+	"github.com/gardener/gardener/pkg/resourcemanager/webhook/seccompprofile"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap/zapcore"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	logzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func TestSeccompProfile(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "SeccompProfile Integration Test Suite")
+}
+
+var (
+	ctx       = context.Background()
+	mgrCancel context.CancelFunc
+
+	logger     logr.Logger
+	testEnv    *envtest.Environment
+	restConfig *rest.Config
+	testClient client.Client
+)
+
+var _ = BeforeSuite(func() {
+	logger = logzap.New(logzap.UseDevMode(true), logzap.WriteTo(GinkgoWriter), logzap.Level(zapcore.Level(1)))
+	logf.SetLogger(logger)
+
+	By("starting test environment")
+	testEnv = &envtest.Environment{
+		WebhookInstallOptions: envtest.WebhookInstallOptions{
+			MutatingWebhooks: getMutatingWebhookConfigurations(),
+		},
+	}
+	var err error
+	restConfig, err = testEnv.Start()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(restConfig).ToNot(BeNil())
+
+	testClient, err = client.New(restConfig, client.Options{Scheme: scheme.Scheme})
+	Expect(err).ToNot(HaveOccurred())
+
+	By("setting up manager")
+	mgr, err := manager.New(restConfig, manager.Options{
+		Port:               testEnv.WebhookInstallOptions.LocalServingPort,
+		Host:               testEnv.WebhookInstallOptions.LocalServingHost,
+		CertDir:            testEnv.WebhookInstallOptions.LocalServingCertDir,
+		MetricsBindAddress: "0",
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("registering webhook")
+	config := seccompprofile.WebhookConfig{Enabled: true}
+	Expect(seccompprofile.AddToManagerWithOptions(mgr, config)).To(Succeed())
+
+	By("starting manager")
+	var mgrContext context.Context
+	mgrContext, mgrCancel = context.WithCancel(ctx)
+	go func() {
+		defer GinkgoRecover()
+		Expect(mgr.Start(mgrContext)).To(Succeed())
+	}()
+})
+
+var _ = AfterSuite(func() {
+	By("stopping manager")
+	mgrCancel()
+
+	By("stopping test environment")
+	Expect(testEnv.Stop()).To(Succeed())
+})
```

Comment:
Nit: In https://github.com/gardener/gardener/pull/6398 @timebertt streamlined the tests:
- to use `DeferCleanup` for stopping the managed and the test env
- introduced a `testID` const to each test that is then set to the logger name

You could check https://github.com/gardener/gardener/pull/6398 and adopt the small changes from there to make sure that the newly introduced env test is streamlined with the existing ones.